scheduler/pkg/agent/rproxy_grpc.go

```diff
@@ -231,6 +225,20 @@ func (rp *reverseGRPCProxy) ModelInfer(ctx context.Context, r *v2.ModelInferRequ
 		}()
 	}()
 
+}
+
+func (rp *reverseGRPCProxy) ModelInfer(ctx context.Context, r *v2.ModelInferRequest) (*v2.ModelInferResponse, error) {
+	logger := rp.logger.WithField("func", "ModelInfer")
+	internalModelName, externalModelName, err := rp.extractModelNamesFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	r.ModelName = internalModelName
+	r.ModelVersion = ""
+
+	// handle scaling metrics
+	syncScalingMetrics(rp, internalModelName, logger)
```

Comment:
based on the comment above I suggest to change the interface to be rp.syncScalingMetrics(internalModelName)` (`rp` has access to logger already)?